loom/graphics/gfxBitmapData.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * ===========================================================================
+ * Loom SDK
+ * Copyright 2011, 2012, 2013
+ * The Game Engine Company, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================
+ */
+
+#pragma once
+
+#include "loom/common/utils/utCommon.h"
+
+namespace GFX
+{
+/*
+ * This class is a utility to load and compare data of images. This is not meant
+ * for general use, but rather as a testing utility.
+ *
+ * BitmapData can be loaded from the current framebuffer or from an asset.
+ * Then this data can be compared and get a numeric result or a diff image can
+ * be generated.
+ */
+class BitmapData
+{
+private:
+    size_t w;
+    size_t h;
+
+    unsigned char* data;
+
+    // Private constructor. Use only static methods for construction.
+    BitmapData(size_t width, size_t height);
+
+public:
+
+    ~BitmapData();
+
+    // Saves the loaded data to a file. This is meant for saving diffs.
```

Comment:
Hmm, that's worth a followup. But I mean - can we save as JPEG, PNG,
RAW...? What library determines it?

On Fri, Sep 4, 2015 at 9:15 AM, Jure Gregorin <notifications@github.com>
wrote:

> In loom/graphics/gfxBitmapData.h
> <https://github.com/LoomSDK/LoomSDK/pull/386#discussion_r38767850>:
>
> > +class BitmapData
> > +{
> > +private:
> > +    size_t w;
> > +    size_t h;
> > +
> > +    unsigned char* data;
> > +
> > +    // Private constructor. Use only static methods for construction.
> > +    BitmapData(size_t width, size_t height);
> > +
> > +public:
> > +
> > +    ~BitmapData();
> > +
> > +    // Saves the loaded data to a file. This is meant for saving diffs.
>
> Currently everything is RGBA, even things supposedly reported RGB (3
> channels) only... O.o
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/386/files#r38767850>.
>